src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -11094,15 +11092,21 @@ private Task WriteUnterminatedSqlValue(object value, MetaType type, int actualLe
                     {
                         Debug.Assert(actualLength == 16, "Invalid length for guid type in com+ object");
                         Span<byte> b = stackalloc byte[16];
-                        SqlGuid sqlGuid = (SqlGuid)value;
-
-                        if (sqlGuid.IsNull)
+                        if (value is Guid guid)
                         {
-                            b.Clear(); // this is needed because initlocals may be supressed in framework assemblies meaning the memory is not automaticaly zeroed
+                            FillGuidBytes(guid, b);
                         }
                         else
```

Comment:
All the previous Guid tests were going through the SqlGuid route which is one of the reasons it was so inefficient. We could add more coverage, hey @lcheunglci I heard you like writing unit tests ... 😁

Reply:
I'll eventually get to TdsParser.  It has 1315 places missing code coverage according to the report.

Reply:
I'd think this would be covered by the Get* and GetfieldValue<*> tests since they're the way to call into this method. I'm not sure how you'd test internal things like TdsParser directly. Unit tests really aren't my thing.